purchase_deposit/wizard/purchase_make_invoice_advance.py

```diff
@@ -0,0 +1,219 @@
+# Copyright 2019 Elico Corp, Dominique K. <dominique.k@elico-corp.com.sg>
+# Copyright 2019 Ecosoft Co., Ltd., Kitti U. <kittiu@ecosoft.co.th>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import time
+from datetime import datetime
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+from odoo.tools import DEFAULT_SERVER_DATETIME_FORMAT
+
+
+class PurchaseAdvancePaymentInv(models.TransientModel):
+    _name = "purchase.advance.payment.inv"
+    _description = "Purchase Advance Payment Invoice"
+
+    advance_payment_method = fields.Selection(
+        [
+            ("percentage", "Down payment (percentage)"),
+            ("fixed", "Deposit payment (fixed amount)"),
+        ],
+        string="What do you want to invoice?",
+        default="percentage",
+        required=True,
+    )
+    purchase_deposit_product_id = fields.Many2one(
+        comodel_name="product.product",
+        string="Deposit Payment Product",
+        domain=[("type", "=", "service")],
+        default=lambda self: self.env.company.purchase_deposit_product_id,
+    )
+    amount = fields.Float(
+        string="Deposit Payment Amount",
+        required=True,
+        help="The amount to be invoiced in advance, taxes excluded.",
+    )
+    deposit_account_id = fields.Many2one(
+        comodel_name="account.account",
+        string="Expense Account",
+        compute="_compute_deposit_account",
+        store=True,
+        readonly=False,
+        domain=[("deprecated", "=", False)],
+        help="Account used for deposits",
+    )
+    deposit_taxes_id = fields.Many2many(
+        comodel_name="account.tax",
+        string="Vendor Taxes",
+        compute="_compute_deposit_account",
+        store=True,
+        readonly=False,
+        help="Taxes used for deposits",
+    )
+
+    @api.depends("purchase_deposit_product_id")
+    def _compute_deposit_account(self):
```

Comment:
Just to follow the convention.

```suggestion
    def _compute_deposit_account_id(self):
```